.github/workflows/backend-unit-tests.yml

```diff
@@ -65,7 +66,28 @@ jobs:
         uses: ./.github/actions/setup-enterprise
         with:
           github-app-name: 'grafana-ci-bot'
+      - run: echo "RESULTS_FILE=$(date --rfc-3339=seconds --utc | sed -s 's/ /-/g')_1.xml" >> $GITHUB_ENV
+      - run: go install github.com/jstemmer/go-junit-report/v2@85bf4716ac1f025f2925510a9f5e9f5bb347c009
       - name: Generate Go code
         run: make gen-go
       - name: Run unit tests
-        run: make test-go-unit
+        run: make test-go-unit GO_TEST_OUTPUT="| tee >(go-junit-report -set-exit-code > $RESULTS_FILE)"
+      - name: Log in to GCS
+        if: github.event_name == 'push' && github.repository == 'grafana/grafana' && always()
+        uses: grafana/shared-workflows/actions/login-to-gcs@login-to-gcs-v0.2.0
+        with:
+          service_account: github-junit-uploader@grafanalabs-workload-identity.iam.gserviceaccount.com
+          bucket: grafana-test-results
+      - if: github.event_name == 'push' && github.repository == 'grafana/grafana' && always()
+        run: echo "BUCKET_PATH=go-unit-tests/$(echo ${REF_NAME} | sed 's/\//-/g')" >> "$GITHUB_ENV"
+        env:
+          REF_NAME: ${{ github.ref_name }}
+      - name: Upload test results
+        if: github.event_name == 'push' && github.repository == 'grafana/grafana' && always()
```

Comment:
What is `&& always()` doing?

Reply:
I'm not sure `&& always()` is what you want. This will make it also run if the workflow is cancelled, whereas `success() || failure()` won't.

Reply:
see also here for some more: https://github.com/orgs/community/discussions/26303#discussioncomment-3251302

Reply:
Yeah the intention is that this runs even if the workflow fails. good suggestion @Proximyst